src/abstractdatatable/abstractdatatable.jl

```diff
@@ -227,7 +227,7 @@ Base.ndims(::AbstractDataTable) = 2
 ##############################################################################
 
 Base.similar(dt::AbstractDataTable, dims::Int) =
-    DataTable(Any[similar(x, dims) for x in columns(dt)], copy(index(dt)))
+    DataTable(Any[similar_nullable(x, dims) for x in columns(dt)], copy(index(dt)))
```

Comment:
What is this change needed for? As of about a month ago we had a `similar_nullable(dt::AbstractDataTable, dims::Int)` function, but I removed it since it wasn't being used. I'm wondering if we should add that back rather than defaulting to `similar_nullable` here.